comp/trace/compression/go.mod

```diff
@@ -0,0 +1,27 @@
+module github.com/DataDog/datadog-agent/comp/trace/compression
```

Comment:
Should we move the Go module from the top level directory? Just create the module if the implementation we want to share.